src/Compilers/CSharp/Test/Emit/Emit/UnmanagedTypeModifierTests.cs

```diff
@@ -0,0 +1,1098 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Emit
+{
+    public class UnmanagedTypeModifierTests : CSharpTestBase
+    {
+        [Fact]
+        public void LoadingADifferentModifierTypeForUnmanagedConstraint()
+        {
+            var ilSource = IsUnmanagedAttributeIL + @"
+.class public auto ansi beforefieldinit TestRef
+       extends [mscorlib]System.Object
+{
+  .method public hidebysig instance void
+          M1<valuetype .ctor (class [mscorlib]System.ValueType modreq([mscorlib]System.Runtime.InteropServices.UnmanagedType)) T>() cil managed
```

Comment:
so it looks like we honor "unmanaged" modreq even if not paired with an attribute. That seems ok to me, just wanted to be sure that this is intentional.

Reply:
@vsadov no we don't. I'll make this test more explicit (it should have the attribute).
Please check other tests that have only the modreq or the attribute and fail at both.